### PR TITLE
added query params validation for /posts with unit testing

### PR DESCRIPTION
**Pull request by zg3d**
## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [x] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description
refactored validation.js to reduce code duplication

added query params validation for /posts with unit testing

fixed testing isssues

## Checklist

- [x] **Quality**: This PR builds and passes our npm test and works locally
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [x] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

File: src/backend/web/validation.js [hunk 1]
```diff
@@ -50,46 +66,44 @@ function validateNewFeed() {
...
-    if (errors.isEmpty()) {
-      return next();
-    }
-    return res.status(400).json({
-      errors: errors.array(),
-    });
-  };
-};
+const queryValidationRules = [
+  // text must be between 1 and 256 and not empty
```
Comment (tonyvugithub) [inline src/backend/web/validation.js:83]: @zg3d : I feel I am asking too many things from you. But can you make one last change to change this to postsQueryValidationRules to make it consistent with the queryValidationRules naming from @chawlapalak ? Greatly appreciate your effort.

**Review by tonyvugithub — CHANGES_REQUESTED** [review]

File: src/backend/web/validation.js (hunk 1 above)
Comment (tonyvugithub) [reply src/backend/web/validation.js:83]: @zg3d : You might have to do another rebase now we found a bug that express-validator was put inside dev-dependencies as supposedly in dependencies. You can see the conversation in Telescope channel

File: src/backend/web/debug.log [hunk 2]
```diff
@@ -0,0 +1 @@
+[1114/201901.723:ERROR:registration_protocol_win.cc(103)] CreateFile: The system cannot find the file specified. (0x2)
```
Comment (tonyvugithub) [inline src/backend/web/debug.log:1]: @zg3d:  why this is in here?

**Review by tonyvugithub — CHANGES_REQUESTED** [review]

File: src/backend/web/debug.log (hunk 2 above)
Comment (zg3d) [reply src/backend/web/debug.log:1]: Im not sure that on my end. I believe that is on the tests end

**Review by tonyvugithub — DISMISSED** [review]

File: src/backend/web/routes/posts.js [hunk 3]
```diff
@@ -3,10 +3,11 @@ const express = require('express');
 const Post = require('../../data/post');
 const { getPosts, getPostsCount } = require('../../utils/storage');
 const { logger } = require('../../utils/logger');
+const { validatePostsQuery } = require('../validation');
 
 const posts = express.Router();
 
-posts.get('/', async (req, res) => {
+posts.get('/', validatePostsQuery, async (req, res) => {
```
Comment (humphd) [inline src/backend/web/routes/posts.js:10]: For consistency, let's always have middleware be accessible via a function call (always returns a new function) vs. a property (which can be altered a runtime):

```js
posts.get('/', validatePostsQuery(), async (req, res) => {
```

File: src/backend/web/routes/query.js [hunk 4]
```diff
@@ -6,7 +6,7 @@ const { search } = require('../../utils/indexer');
 
 const router = express.Router();
 
-router.get('/', validateQuery(), async (req, res) => {
+router.get('/', validateQuery, async (req, res) => {
```
Comment (humphd) [inline src/backend/web/routes/query.js:9]: Same deal here, prefer function call.

**Review by humphd — CHANGES_REQUESTED** [review]
One small thing, then this is good to go.

We'll have to make sure the other PRs that are doing similar work get rebased and use this new helper function.

**Comment by zg3d** [comment]
I added debug.log to gitignore if thats okay

**Review by humphd — APPROVED** [review]
Awesome!  This looks excellent.

**Review by tonyvugithub — APPROVED** [review]

**Comment by humphd** [comment]
cc @eunbeek, @sonechca, @sanji11, @chawlapalak .  This PR just landed and it changes what needs to happen a bit in #1359, #1363, and #1357.  Please rebase your work on `master` to pick it up, and follow @zg3d's lead from this work.

@zg3d, @tonyvugithub, @chawlapalak: if you can help me with reviews on those others so we don't mess up what was started here, that would be helpful.

**Pull request merged**